Source: Timo Gurr <tgurr@exherbo.org>
Upstream: Reported, http://bugs.mysql.com/bug.php?id=79229
Reason: Fix starting mysqld via the provided systemd service file

diff -Naur mysql-5.7.9/scripts/systemd/mysqld.service.in mysql-5.7.9.new/scripts/systemd/mysqld.service.in
--- mysql-5.7.9/scripts/systemd/mysqld.service.in	2015-10-12 08:21:33.000000000 +0200
+++ mysql-5.7.9.new/scripts/systemd/mysqld.service.in	2015-11-11 15:06:10.662347519 +0100
@@ -42,7 +42,7 @@
 ExecStartPre=@bindir@/mysqld_pre_systemd
 
 # Start main service
-ExecStart=@libexecdir@/mysqld --daemonize $MYSQLD_OPTS
+ExecStart=@libexecdir@/mysqld --daemonize --pid-file=@SYSTEMD_PID_DIR@/mysqld.pid $MYSQLD_OPTS
 
 # Use this to switch malloc implementation
 EnvironmentFile=-/etc/sysconfig/mysql
